A couple of bug fixes and updates of the Debian package

schacht-certat opened the pull request:
d/changelog:
  * Fix bogus-mail-host-in-debian-changelog
    https://lintian.debian.org/tags/bogus-mail-host-in-debian-changelog.html
  * Fix depends-on-essential-package-without-using-version
    This referres to `debianutils`. See
    https://lintian.debian.org/tags/build-depends-on-essential-package-without-using-version.html
    "The only reason to list an explicit dependency on an essential package
    is if you need a particular version of that package"
  * Fix: description-starts-with-package-name
    https://lintian.debian.org/tags/description-starts-with-package-name.html
    This also adds a synopsis, because according to policy, a Debian
    package description consists 'of two parts, the synopsis or the
    short description, and the long description'
  * Fix: syntax-error-in-debian-changelog
    There was a changed-by line missing, this got somehow introduced in
    eb6b38091d865f5e36920df57984eec638fed7a0; not a big deal, but fixing
    it makes lintian happy.
 d/control:
  * Run wrap-and-sort on d/control
  * Use sphinx and the sphinx rdt theme to build and manage
    documentation. This adds the necessary dependencies to the control
    file
  * Replaces the old style debian/compat file with a build-dependency on
    debhelper-compat.
 d/rules:
  * Enable the sphinx debhelper
    The --with systemd argument for debhelper is no longer provided in
    compat >= 11 - Debian packages use dh_installsystemd nowadays
    This also removes the systemd override in d/rules, because this is
    now default behaviour since compat 10.

wagner-certat commented on debian/control:
```diff
@@ -2,22 +2,56 @@ Source: intelmq
 Section: python
 Priority: optional
-Build-Depends: debhelper (>= 4.1.16), python3-all, python3-setuptools, quilt, dh-python, python-setuptools, dh-systemd, safe-rm, python3-requests, python3-redis, python3-dnspython, python3-psutil, python3-dateutil, python3-termstyle, python3-tz, lsb-release
+Build-Depends: debhelper-compat (= 13),
```
With that requirement, we cannot build the package anymore on Ubuntu 16.04, 18.04, 20.04 and Debian 9, 10.

schacht-certat replied:
I've changed that to `debhelper-compat (=10)` which should work for all of those, backports considered

wagner-certat commented on debian/control:
```diff
@@ -2,22 +2,56 @@ Source: intelmq
 Section: python
 Priority: optional
-Build-Depends: debhelper (>= 4.1.16), python3-all, python3-setuptools, quilt, dh-python, python-setuptools, dh-systemd, safe-rm, python3-requests, python3-redis, python3-dnspython, python3-psutil, python3-dateutil, python3-termstyle, python3-tz, lsb-release
+Build-Depends: debhelper-compat (= 10),
```
That doesn't work with Debian 9.0 and Ubuntu 16.04 now. If we gonna drop support for them, as they use python 3.5, that would be fine.

schacht-certat replied:
Oke, reverted to `9`

wagner-certat commented:
I've update the unstable repo to commit 1cc4f11 now to get all the build results: https://build.opensuse.org/package/show/home:sebix:intelmq:unstable/intelmq

schacht-certat commented:
I have reverted the change regarding `debhelper-compat` and added some more build-dependencies. The build should work on Debian 9 now, the only thing one has to do is to install `python3-cerberus` by hand.

wagner-certat reviewed (changes requested):
The Debian build fails on travis: https://travis-ci.org/github/certtools/intelmq/jobs/747603328#L501

> The build should work on Debian 9 now, the only thing one has to do is to install `python3-cerberus` by hand.

Cerberus is only needed at build time for the tests, not at runtime. So it shouldn't be an issue

I updated the unstable package to commit 01af64fcc: https://build.opensuse.org/package/show/home:sebix:intelmq:unstable/intelmq

wagner-certat commented:
> The Debian build fails on travis: https://travis-ci.org/github/certtools/intelmq/jobs/747603328#L501

Maybe using GitHub Actions, like for intelmq-api, works around this issue? I don't see this failure on OBS.
While Travis uses the command `dpkg-buildpackage -us -uc -d`, the GitHub Action uses `dpkg-buildpackage -us -uc -b`
